File: src/main/java/com/hackerrank/ClimbingTheLeaderboard.java
package com.hackerrank;

import java.util.Scanner;
import java.util.Stack;

public class ClimbingTheLeaderboard {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();

        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < n; i++) {
            int score = in.nextInt();
            if (i == 0 || stack.peek() != score) {
                stack.push(score);
            }
        }

        int m = in.nextInt();
        int[] alice = new int[m];
        for (int i = 0; i < m; i++) {
            alice[i] = in.nextInt();

            while (!stack.empty() && stack.peek() <= alice[i]) {
                stack.pop();
            }
            System.out.println(stack.isEmpty() ? "1" : stack.size() + 1);
        }
    }
}

File: src/main/kotlin/com/hackerrank/ClimbingTheLeaderboard.kt
package com.hackerrank

import java.util.*

fun main(args: Array<String>) {
    val scan = Scanner(System.`in`)

    val n = scan.nextInt()
    val scores = Stack<Int>()
    for (i in 0..n - 1) {
        val score = scan.nextInt()
        if (i == 0 || scores.peek() != score) {
            scores.push(score)
        }
    }

    val m = scan.nextInt()
    for (i in 0..m - 1) {
        val alice = scan.nextInt()
        while (!scores.empty() && scores.peek() <= alice) {
            scores.pop()
        }
        println(if (scores.isEmpty()) "1" else scores.size + 1)
    }
}